### PR TITLE
use bytemuck for disk bucket restart

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Speeding up restart.
Reusing disk index files on start.

#### Summary of Changes
Use bytemuck for manipulations.
Improve test.

Fixes #

### CODE REVIEW

File: bucket_map/src/restart.rs [hunk 1]
```diff
@@ -16,21 +17,27 @@ use {
...
     max_search: u8,
     /// padding to get header to u128 aligned
     _dummy: [u8; 15],
 }
 
-#[derive(Debug)]
+// In order to safely guarantee Header is Pod, it cannot have any padding.
+const _: () = assert!(
+    std::mem::size_of::<Header>() == std::mem::size_of::<u128>() * 2,
+    "Header cannot have any padding"
```
Comment (HaoranYi) [inline bucket_map/src/restart.rs:37]: nit: the error message is not accurate.

File: bucket_map/src/restart.rs [hunk 2]
```diff
@@ -41,6 +48,12 @@ pub(crate) struct OneIndexBucket {
     _dummy: u64,
 }
 
+// In order to safely guarantee Header is Pod, it cannot have any padding.
+const _: () = assert!(
+    std::mem::size_of::<OneIndexBucket>() == std::mem::size_of::<u128>() * 2,
+    "Header cannot have any padding"
```
Comment (HaoranYi) [inline bucket_map/src/restart.rs:54]: nit: inaccurate error message.

**Review by HaoranYi — APPROVED** [review]
lgtm

File: bucket_map/src/restart.rs (hunk 1 above)
Comment (jeffwashington) [reply bucket_map/src/restart.rs:37]: I suppose it could be more clear if I said no automatic padding can be added to the header. Thoughts?

Comment (HaoranYi) [reply bucket_map/src/restart.rs:37]: Maybe, just say "incorrect size of header struct" generally.

Comment (jeffwashington) [reply bucket_map/src/restart.rs:37]: done in next pr

**Pull request merged**